AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -934,66 +934,39 @@ function Request-PowerShellGetModule {
-            Get-Module -Name AzureAD -ListAvailable 
+    foreach ($module in $requiredModules) {
+        $installedModule = Get-Module -Name $module -ListAvailable
+        if ($null -eq $installedModule) {
+            Write-Host "Missing module: $module"
+            $missingModules += $module
         }
     }
 
-    if (
-        ($PSVersionTable.PSVersion -gt [Version]::new(6,0,0,0) -and $null -eq $winCompat) -or 
-        $null -eq $azureADModule
-    ) {
-        $caption = "Install AzureAD PowerShell module"
-        $verboseConfirmMessage = "This cmdlet requires the Azure AD PowerShell module. This can be automatically installed now if you are running in an elevated sessions."
+    if ($missingModules.Count -gt 0) {
+        $caption = "Install missing Microsoft.Graph PowerShell modules"
+        $verboseConfirmMessage = "This cmdlet requires the Microsoft.Graph PowerShell modules. The missing ones can be automatically installed now if you are running in an elevated sessions."
         
         if ($PSCmdlet.ShouldProcess($verboseConfirmMessage, $verboseConfirmMessage, $caption)) {
             if (!(Get-IsElevatedSession)) {
                 Write-Error `
-                        -Message "To install AzureAD, you must run this cmdlet as an administrator. This cmdlet may not generally require administrator privileges." `
-                        -ErrorAction Stop
-            }
-
-            if ($PSVersionTable.PSVersion -gt [Version]::new(6,0,0) -and $null -eq $winCompat) {
-                Install-Module `
-                        -Name WindowsCompatibility `
-                        -Repository PSGallery `
-                        -AllowClobber `
-                        -Force `
+                        -Message "To install the missing Microsoft.Graph modules, you must run this cmdlet as an administrator. This cmdlet may not generally require administrator privileges." `
                         -ErrorAction Stop
-
-                Import-Module -Name WindowsCompatibility
             }
             
-            $scriptBlock = { 
-                $azureADModule = Get-Module -Name AzureAD -ListAvailable
-                if ($null -eq $azureADModule) {
-                    Install-Module `
-                            -Name AzureAD `
-                            -Repository PSGallery `
-                            -AllowClobber `
-                            -Force `
-                            -ErrorAction Stop
-                }
-            }
-
-            if ($PSVersionTable.PSVersion -gt [Version]::new(6,0,0)) {
-                Invoke-WinCommand `
-                        -ScriptBlock $scriptBlock `
-                        -Verbose:$false `
-                        -ErrorAction Stop
-            } else {
-                $scriptBlock.Invoke()
-            }
+            Write-Host "Installing missing modules: $($missingModules -join ', ')"
+            Install-Module `
+                -Name $missingModules `

```

Comment:
Didn't know that you could instill a `String[]` of modules actually, but that's cool!

Reply:
https://learn.microsoft.com/en-us/powershell/module/powershellget/install-module?view=powershellget-2.x#:~:text=Install%2DModule%0A%20%20%20%20%20%20%20%5B%2DName,WhatIf%5D%0A%20%20%20%20%20%20%20%5B%2DConfirm%5D%0A%20%20%20%20%20%20%20%5B%3CCommonParameters%3E%5D

Yes, link to the doc